plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Core.hs

```diff
@@ -1,7 +1,27 @@
-module UntypedPlutusCore.Core
-    ( module Export
-    ) where
+{-# LANGUAGE LambdaCase #-}
+
+module UntypedPlutusCore.Core (
+  module Export,
+  splitLamAbs,
+  splitApplication,
+) where
 
 import UntypedPlutusCore.Core.Instance as Export
 import UntypedPlutusCore.Core.Plated as Export
 import UntypedPlutusCore.Core.Type as Export
+
+import Data.Bifunctor
+
+-- | Strips off lambda binders.
+splitLamAbs :: Term name uni fun a -> ([name], Term name uni fun a)
```

Comment:
See below comment, this is similar to [computeArity](https://github.com/input-output-hk/plutus/blob/5b7848040e932f3344ec31ff131f52371470da72/plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/CallSiteInline.hs#L109) maybe add a comment and rename one of them so they are more clearly doing something similar.